test/src/main/java/org/apache/accumulo/test/ample/TestAmple.java

```diff
@@ -0,0 +1,212 @@
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.ServerColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataTime;
+import org.apache.accumulo.core.metadata.schema.TabletsMetadata;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.util.ColumnFQ;
+import org.apache.accumulo.manager.Manager;
+import org.apache.accumulo.miniclusterImpl.MiniAccumuloClusterImpl;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.metadata.ConditionalTabletsMutatorImpl;
+import org.apache.accumulo.server.metadata.ServerAmpleImpl;
+import org.apache.accumulo.server.metadata.TabletsMutatorImpl;
+import org.apache.hadoop.io.Text;
+import org.easymock.EasyMock;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.MoreCollectors;
+
+public class TestAmple {
+
+  public static Ample create(ServerContext context, Map<DataLevel,String> tables) {
+    return new TestServerAmpleImpl(context, tables);
+  }
+
+  public static class TestServerAmpleImpl extends ServerAmpleImpl {
+
+    private final Map<DataLevel,String> tables;
+
+    public TestServerAmpleImpl(ServerContext context, final Map<DataLevel,String> tables) {
+      super(context);
+      this.tables = Map.copyOf(tables);
+      Preconditions.checkArgument(tables.containsKey(DataLevel.USER));
+    }
+
+    @Override
+    public TabletsMutator mutateTablets() {
+      return new TabletsMutatorImpl(getContext()) {
+        @Override
+        protected String getMetadataTableName() {
+          return TestServerAmpleImpl.this.getMetadataTableName();
+        }
+      };
+    }
+
+    @Override
+    public TabletsMetadata.TableOptions readTablets() {
+      return TabletsMetadata.builder(getContext(), getMetadataTableName());
+    }
+
+    /**
+     * Create default metadata for a Table
+     *
+     * TODO: Add a way to pass in options for config
+     *
+     * @param tableId The id of the table to create metadata for
+     */
+    public void createMetadata(TableId tableId) {
+      try (var tabletsMutator = mutateTablets()) {
+        var extent = new KeyExtent(tableId, null, null);
+        var tabletMutator = tabletsMutator.mutateTablet(extent);
+        String dirName = ServerColumnFamily.DEFAULT_TABLET_DIR_NAME;
+        tabletMutator.putPrevEndRow(extent.prevEndRow());
+        tabletMutator.putDirName(dirName);
+        tabletMutator.putTime(new MetadataTime(0, TimeType.MILLIS));
+        tabletMutator.putTabletAvailability(TabletAvailability.HOSTED);
+        tabletMutator.mutate();
+      } catch (Exception e) {
+        throw new IllegalStateException(e);
+      }
+    }
+
+    /**
+     * Create metadata for a table by copying existing metadata for the table from the metadata
+     * table in an existing Accumulo instance
+     *
+     * TODO: Add config parents (such as a way to include/exclude what is copied, etc)
```

Comment:
Could pass a `BiPredicate<Key, Value>` to the function.  Not sure if I have ever used BiPredicate, this seems like the perfect place for it.

Reply:
I like this idea, I just pushed up a new commit that implements it https://github.com/apache/accumulo/pull/4415/commits/e6b199eed5cf2ae1dc20a2d7873a2e49843bb964